test/e2e/fleet_test.go

```diff
@@ -43,6 +53,133 @@ func TestCreateFleetAndAllocate(t *testing.T) {
 	assert.Equal(t, v1alpha1.Allocated, fa.Status.GameServer.Status.State)
 }
 
+func TestScaleFleetUpAndDownWithAllocation(t *testing.T) {
+	t.Parallel()
+	alpha1 := framework.AgonesClient.StableV1alpha1()
+
+	flt := defaultFleet()
+	flt.Spec.Replicas = 1
+	flt, err := alpha1.Fleets(defaultNs).Create(flt)
+	if assert.Nil(t, err) {
+		defer alpha1.Fleets(defaultNs).Delete(flt.ObjectMeta.Name, nil) // nolint:errcheck
+	}
+
+	assert.Equal(t, int32(1), flt.Spec.Replicas)
+
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+	assert.Nil(t, err, "fleet not ready")
+
+	// scale up
+	flt, err = scaleFleet(flt, 3)
+	assert.Nil(t, err)
+	assert.Equal(t, int32(3), flt.Spec.Replicas)
+
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+	assert.Nil(t, err)
+
+	// get an allocation
+	fa := &v1alpha1.FleetAllocation{
+		ObjectMeta: metav1.ObjectMeta{GenerateName: "allocation-", Namespace: defaultNs},
+		Spec: v1alpha1.FleetAllocationSpec{
+			FleetName: flt.ObjectMeta.Name,
+		},
+	}
+
+	fa, err = alpha1.FleetAllocations(defaultNs).Create(fa)
+	assert.Nil(t, err)
+	assert.Equal(t, v1alpha1.Allocated, fa.Status.GameServer.Status.State)
+	err = framework.WaitForFleetCondition(flt, func(fleet *v1alpha1.Fleet) bool {
+		return fleet.Status.AllocatedReplicas == 1
+	})
+	assert.Nil(t, err)
+
+	// scale down, with allocation
+	flt, err = scaleFleet(flt, 1)
+	assert.Nil(t, err)
+	err = framework.WaitForFleetCondition(flt, e2e.FleetReadyCount(0))
+	assert.Nil(t, err)
+
+	// remove allocation
+	gp := int64(1)
+	err = alpha1.GameServers(defaultNs).Delete(fa.Status.GameServer.ObjectMeta.Name, &metav1.DeleteOptions{GracePeriodSeconds: &gp})
```

Comment:
Comment says remove allocation but I (mis)understand it deletes a game server. Am I right?

Reply:
Good point - it's deleting the allocated game server. Changing the comment to be more clear.